src/CADL.Extension/Emitter.Csharp/src/lib/model.ts

```diff
@@ -107,11 +110,23 @@ export function mapCadlTypeToCSharpInputTypeKind(
 
 function getCSharpInputTypeKindByIntrinsicModelName(
     name: string,
-    format?: string
+    format?: string,
+    encode?: EncodeData
 ): InputTypeKind {
     switch (name) {
         case "bytes":
-            return InputTypeKind.Bytes;
+            switch (encode?.encoding) {
+                case undefined:
+                case "base64":
+                    return InputTypeKind.Bytes;
+                case "base64":
```

Comment:
Could you explain why use the same condition of ‘base64’ in both line 120 & 122?